src/rust/engine/process_execution/src/local.rs

```diff
@@ -170,10 +214,10 @@ impl super::CommandRunner for CommandRunner {
         };
 
         output_snapshot
-          .map(|snapshot| FallibleExecuteProcessResult {
-            stdout: Bytes::from(output.stdout),
-            stderr: Bytes::from(output.stderr),
-            exit_code: output.status.code().unwrap(),
+          .map(move |snapshot| FallibleExecuteProcessResult {
+            stdout: stdout.freeze(),
+            stderr: stderr.freeze(),
+            exit_code: exit_code.unwrap_or(-1),
```

Comment:
I would expect there to always be an exit code, right? Maybe make this an `exit_code.ok_or_else(|| "Process had no exit code".to_owned())`

Reply:
https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code